app/models/katello/content_view_puppet_module.rb

```diff
@@ -8,6 +8,7 @@ class ContentViewPuppetModule < Katello::Model
     validates :content_view_id, :presence => true
     validates :name, :uniqueness => { :scope => :content_view_id }, :allow_blank => true
     validates :uuid, :uniqueness => { :scope => :content_view_id }, :allow_blank => true
+    validate :uuid_associated_with_puppet_module
```

Comment:
Honestly, `uuid` is redundant information; The same goes for the `name` and `author` columns. Usually join tables contain only the two `id` columns, and additional columns in the join table usually have something to do with the relationship itself. Can you justify the current design?

If it is true that the `uuid`, `name`, and `author` columns are always identical (and thus redundant), then we should validate that, in which case I should add additional checks to this validation for `name` and `author`.

It's also very strange, from a Rails idioms perspective, that `content_view.puppet_modules` returns a collection of `ContentViewPuppetModules` rather than `PuppetModules`.